src/main/java/duke/helpers/Parser.java

```diff
@@ -0,0 +1,117 @@
+package duke.helpers;
+
+import duke.exceptions.OutOfIndexException;
+import duke.exceptions.VagueInputException;
+import duke.exceptions.WrongBooleanException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.io.PrintWriter;
+import java.util.Scanner;
+
+/**
+ * Handles input cases by the user.
+ */
+public class Parser {
+
+    /**
+     * Handles inputs, and outputs the according string.
+     *
+     * @param sc Scans the inputs of the user.
+     * @param tasks Stores the Tasks objects into
+     * @param pw Writes to the screen for the user to comprehend.
+     */
+    public static void handleInputs(Scanner sc, TaskList tasks, PrintWriter pw) {
+        String input = sc.nextLine();
+        while (!input.equals("bye")) {
+            String[] splitArr = input.split(" ");
+            try {
+                if (input.equals("list")) {
+                    System.out.println(Ui.formatStr(tasks.listThings()));
```

Comment:
Maybe switch can be used instead of if else here? It would increase readability and i think it's faster.